pkg/login/social/common.go

```diff
@@ -71,23 +70,24 @@ func HttpGet(client *http.Client, url string) (response HttpGetResponse, err err
 }
 
 func (s *SocialBase) searchJSONForAttr(attributePath string, data []byte) (string, error) {
-	if attributePath == "" {
-		return "", errors.New("no attribute path specified")
-	}
-	if len(data) == 0 {
-		return "", errors.New("empty user info JSON response provided")
+	if attributePath == "" || len(data) == 0 {
+		return "", nil
 	}
+
 	var buf interface{}
 	if err := json.Unmarshal(data, &buf); err != nil {
 		return "", errutil.Wrap("failed to unmarshal user info JSON response", err)
 	}
+
 	val, err := jmespath.Search(attributePath, buf)
 	if err != nil {
 		return "", errutil.Wrapf(err, "failed to search user info JSON response with provided path: %q", attributePath)
```

Comment:
Can this error be triggered by bad user params?

Reply:
From a bad server configuration or if auth provider returns non-valid json maybe

Reply:
In that case, it makes sense to return an error so it gets dealt with (either by logging or bubling).